native/src/main/java/io/ballerina/stdlib/data/jsondata/json/JsonTraverse.java

```diff
@@ -0,0 +1,309 @@
+                        traverseArrayMembers(expectedArraySize, array, elementType, currentJsonNode);
+                    }
+                }
+                case TypeTags.TUPLE_TAG -> {
+                    TupleType tupleType = (TupleType) rootArray;
+                    Type restType = tupleType.getRestType();
+                    int expectedTupleTypeCount = tupleType.getTupleTypes().size();
+                    if (expectedTupleTypeCount > array.getLength()) {
+                        throw DiagnosticLog.error(DiagnosticErrorCode.ARRAY_SIZE_MISMATCH);
+                    }
+                    for (int i = 0; i < array.getLength(); i++) {
+                        Object jsonMember = array.get(i);
+                        Object nextJsonNode;
+                        if (i < expectedTupleTypeCount) {
+                            nextJsonNode = traverseJson(jsonMember, tupleType.getTupleTypes().get(i));
+                        } else if (restType != null) {
+                            nextJsonNode = traverseJson(jsonMember, restType);
+                        } else if (!allowDataProjection) {
+                            throw DiagnosticLog.error(DiagnosticErrorCode.ARRAY_SIZE_MISMATCH);
+                        } else {
+                            continue;
+                        }
+                        ((BArray) currentJsonNode).add(i, nextJsonNode);
+                    }
+                }
+            }
+            return currentJsonNode;
+        }
+
+        private void traverseArrayMembers(long length, BArray array, Type elementType, Object currentJsonNode) {
+            for (int i = 0; i < length; i++) {
+                ((BArray) currentJsonNode).add(i, traverseJson(array.get(i), elementType));
+            }
+        }
+
+        private void addRestField(Type restFieldType, BString key, Object jsonMember, Object currentJsonNode) {
+            Object nextJsonValue;
+            switch (restFieldType.getTag()) {
+                case TypeTags.ANYDATA_TAG, TypeTags.JSON_TAG ->
+                        ((BMap<BString, Object>) currentJsonNode).put(key, jsonMember);
+                case TypeTags.BOOLEAN_TAG, TypeTags.INT_TAG, TypeTags.FLOAT_TAG, TypeTags.DECIMAL_TAG,
+                        TypeTags.STRING_TAG -> {
+                    if (checkTypeCompatibility(restFieldType, jsonMember)) {
+                        ((BMap<BString, Object>) currentJsonNode).put(key, jsonMember);
+                    }
+                }
+                default -> {
+                    nextJsonValue = traverseJson(jsonMember, restFieldType);
+                    ((BMap<BString, Object>) currentJsonNode).put(key, nextJsonValue);
+                }
+            }
+        }
+
+        private boolean checkTypeCompatibility(Type type, Object json) {
+            return ((json == null && type.getTag() == TypeTags.NULL_TAG)
+                    || (json instanceof BString && type.getTag() == TypeTags.STRING_TAG)
```

Comment:
If the type is anydata, this should return true?
wdyt?

Reply:
Actually we have handle those for restType here 
https://github.com/prakanth97/module-ballerina-data.jsondata/blob/bda32fb8916be0c4b83763939b3dac5b7a1fba42/native/src/main/java/io/ballerina/stdlib/data/jsondata/json/JsonTraverse.java#L248 
 
 other scenarios here 
 https://github.com/prakanth97/module-ballerina-data.jsondata/blob/bda32fb8916be0c4b83763939b3dac5b7a1fba42/native/src/main/java/io/ballerina/stdlib/data/jsondata/json/JsonTraverse.java#L121